File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
import sys
class Solution(object):
    def findGCD(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mx=-sys.maxint-1
        mn=sys.maxint
        div=1
        for i in nums:
            if i<mn:
                mn=i
            if i>mx:
                mx=i
        for i in range(mn*mx,0,-1):
            if mx%i==0 and mn%i==0:
                div=i
                break
        return div
            
            

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.c

int findGCD(int* nums, int numsSize){
    int max = nums[0];
    int min = nums[numsSize-1];
    
    for(int i=0;i<numsSize;i++){
        if(nums[i] > max){max = nums[i];}
        if(nums[i] < min){min = nums[i];}
    }
    
    //printf("max is %d and min is %d\n",max,min);
    int gcd = 1;
    
    for(int i=2;i<=max;i++){
        if( (max % i==0) && (min % i ==0)){
            gcd = i;
        }
    }
    return gcd;
}